### PR TITLE
add POST/DELETE tags to collection for tagIds. add PUT details for the typeIds/tagGroupIds

### PR DESCRIPTION
**Pull request by imans777**

### CODE REVIEW

File: lib/collection.model.js [hunk 1]
```diff
@@ -35,168 +35,230 @@ class Collection extends Base {
...
+     * NOTE: this body will overwrite the previous typeIds and tagGroupIds!
+     * @returns {Promise.<*>}
+     */
+    updateDetails(cid, body) {
+        if (!mongoose.Types.ObjectId(cid)) return Promise.reject(error.collectionIdIsNotValid);
+
+        return this.CollectionModel.update({_id: cid}, body);
+    }
+
+    //TODO: not sure if this is needed :-?
```
Comment (eabasir) [inline lib/collection.model.js]: remove comments

File: routes/api.js [hunk 2]
```diff
@@ -161,17 +161,20 @@ router.use('/product/image/:id/:colorId', function (req, res, next) {
...
 router.delete('/collection/product/:cid/:pid', apiResponse('Collection', 'deleteProductFromCollection', false, ['params']));
-router.put('/collection/product/:cid/:pid', apiResponse('Collection', 'setProductToCollection', false, ['params']));
+router.post('/collection/product/:cid/:pid', apiResponse('Collection', 'setProductToCollection', false, ['params']));
+router.delete('/collection/tag/:cid/:tid', apiResponse('Collection', 'deleteTagFromCollection', false, ['params']));
+router.post('/collection/tag/:cid/:tid', apiResponse('Collection', 'setTagToCollection', false, ['params']));
+router.put('/collection/detail/:cid', apiResponse('Collection', 'updateDetails', false, ['params.cid', 'body']));
 router.put('/collection', apiResponse('Collection', 'setCollection', false, ['body']));
-router.get('/collection/products/:cid', apiResponse('Collection', 'getProductsFromCollection', false, ['params.cid']));
+//TODO: I think we don't need this :-?
+// router.get('/collection/products/:cid', apiResponse('Collection', 'getProductsFromCollection', false, ['params.cid']));
```
Comment (eabasir) [inline routes/api.js]: what is this going to done? this get products of a collection? so we want it in  products page when user click on a collection.

File: spec/api/collection/delete.spec.js [hunk 3]
```diff
@@ -100,6 +95,26 @@ describe('DELETE Collection', () => {
     }).catch(lib.helpers.errorHandler.bind(this));
   });
 
+  it('should delete tag from collection', function (done) {
```
Comment (eabasir) [inline spec/api/collection/delete.spec.js]: only admin can delete tags from a collection. use addAndLoginAgent function inside dbHelper  in beforeEach to add an Admin agent and use its cookie to send within all request

File: spec/api/collection/post.spec.js [hunk 4]
```diff
@@ -1,38 +1,112 @@
 const models = require('../../../mongo/models.mongo');
```
Comment (eabasir) [inline spec/api/collection/post.spec.js]: same as delete tests. only admin can do these functionalities

File: spec/api/collection/put.spec.js [hunk 5]
```diff
@@ -5,21 +5,34 @@ const error = require('../../../lib/errors.list');
 const mongoose = require('mongoose');
```
Comment (eabasir) [inline spec/api/collection/put.spec.js]: same as delete tests. only admin can do these functionalities

File: routes/api.js (hunk 2 above)
Comment (imans777) [reply routes/api.js]: Yes, but we already get all the products in "GET /collection/:cid"! So I thought we don't need this anymore, as we're doing the same thing there :-?

**Pull request merged**